packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts

```diff
@@ -603,6 +593,68 @@ export class FieldMetadataService extends TypeOrmQueryService<FieldMetadataEntit
     }
   }
 
+  private async validateFieldMetadataOptions({
+    fieldMetadataInput,
+    fieldMetadataType,
+  }: {
+    fieldMetadataInput: CreateFieldInput | UpdateFieldInput;
+    fieldMetadataType: FieldMetadataType;
+  }) {
+    if (!isEnumFieldMetadataType(fieldMetadataType)) {
+      return;
+    }
+
+    const { options } = fieldMetadataInput;
+    if (!isDefined(options) || options.length === 0) {
+      throw new FieldMetadataException(
+        'Options are required for enum fields',
+        FieldMetadataExceptionCode.INVALID_FIELD_INPUT,
+      );
+    }
+
+    const seenOptionIds = new Set<string>();
+    for (const option of options) {
+      if (!isDefined(option.id)) {
+        throw new FieldMetadataException(
+          'Option id is required',
+          FieldMetadataExceptionCode.INVALID_FIELD_INPUT,
+        );
+      }
+
+      const parseResult = z.string().uuid().safeParse(option.id);
```

Comment:
@Weiko what do you think, do we slowly move to zod? do we accept both zod and classvalidator?

Reply:
Reached out to both you and @Weiko to discuss a potential shared zod validation schema between front and back